x-pack/auditbeat/processors/sessionmd/add_session_metadata.go

```diff
@@ -145,13 +160,24 @@ func (p *addSessionMetadata) enrich(ev *beat.Event) (*beat.Event, error) {
 		return nil, fmt.Errorf("cannot parse pid field '%s': %w", p.config.PIDField, err)
 	}
 
-	fullProcess, err := p.db.GetProcess(pid)
-	if err != nil {
-		e := fmt.Errorf("pid %v not found in db: %w", pid, err)
-		p.logger.Errorf("%v", e)
-		return nil, e
+	var fullProcess types.Process
+	if p.backend == "kernel_tracing" {
+		// kernel_tracing doesn't enrich with the processor DB;  process info is taken directly from quark cache
+		proc, err := p.provider.GetProcess(pid)
+		if err != nil {
+			e := fmt.Errorf("pid %v not found in db: %w", pid, err)
+			p.logger.Errorf("%v", e)
```

Comment:
Is this being double logged because it is both logged here and returned to the caller?

At worst, I would say this is a `Warnf`. This could be for a future improvement, but I think this might be better suited to a metric that we increment for PID misses.

Reply:
This is because Processor code isn't logging the error at all when returned to it, so it needs to be logged here. I'll changed it to `Warnw`